specs/eventing/data-plane.md

```diff
@@ -1,125 +1,184 @@
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in
+[RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
+When not specified in this document, the
+[CloudEvents HTTP bindings, version 1.0](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+and [HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards MUST be
+followed (with the CloudEvents bindings taking precedence in the case of
+conflict).
 
-### Retries
+The current version of this document does not describe protocol negotiation or
+any delivery mechanism other than HTTP 1.1. Future versions might define
+protocol negotiation to optimize delivery; compliant implementations SHOULD aim
+to interoperate by ignoring unrecognized negotiation options (such as
+[HTTP `Upgrade` headers](https://datatracker.ietf.org/doc/html/rfc7230#section-6.7)).
 
-Sinks should expect that retries and accept possibility that duplicate events
-may be delivered.
+## Event Delivery
 
-### Error handling
+### Minimum supported protocol
 
-If Sink is not returning HTTP success header (200 or 202) then the event may be
-sent again. If the event can not be delivered then some sources of events (such
-as Knative sources, brokers or channels) MAY support
-[dead letter sink or channel](https://github.com/knative/eventing/blob/main/docs/delivery/README.md) for events that can not be
-delivered.
+All senders and recipients MUST support the CloudEvents 1.0 protocol and the
+[binary](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[structured](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+content modes of the CloudEvents HTTP binding. Senders which do not advertise
+the ability to accept [reply events](#derived-reply-events) MAY implement only
+one content mode, as the recipient is not allowed to negotiate the content mode.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+MUST be performed using the CloudEvents HTTP Binding, version 1.x.
+
+Senders MAY probe the recipient with an
+[HTTP OPTIONS request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if
+implemented, the recipient MUST indicate support for the POST verb using the
+[`Allow` header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders
+which receive an error when probing with HTTP OPTIONS SHOULD proceed using the
+HTTP POST mechanism.
+
+### Event Acknowledgement and Delivery Retry
```

Comment:
@evankanderson once the new spec is merged can you open a docs issue to add the response codes to knative.dev? I think this would be useful info to doc if it's standardized now

Reply:
Sure